07.ls_object/display.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class Display
+  COLUMN_NUMBER = 3
+
+  def initialize(file_stats)
+    @file_stats = file_stats
+  end
+
+  def print_simple_list
+    row_number = (@file_stats.size.to_f / COLUMN_NUMBER).ceil
+
+    max_lengths = @file_stats.each_slice(row_number).map do |file_stat|
+      file_stat.map { |file| file.name.size }.max
+    end
+
+    before_format_display = @file_stats.map(&:name).each_slice(row_number).to_a
+    before_format_display.last.size == row_number || before_format_display.last.fill(nil, before_format_display.last.size...row_number)
+    formatted_display = before_format_display.transpose
+
+    formatted_display.each do |display|
+      display.each_with_index do |file, index|
+        print file.ljust(max_lengths[index] + 2) unless file.nil?
+      end
+      puts
+    end
+  end
+
+  def print_detailed_list
+    max_nlink = @file_stats.map { |file| file.nlink.to_s.size }.max
+    max_user = @file_stats.map { |file| file.user.size }.max
+    max_group = @file_stats.map { |file| file.group.size }.max
+    max_size = @file_stats.map { |file| file.bytesize.to_s.size }.max
+
+    puts "合計 #{@file_stats.sum(&:blocks) / 2}"
+    @file_stats.each do |file|
+      puts [
+        file.mode,
+        file.nlink.to_s.rjust(max_nlink),
+        file.user.ljust(max_user),
+        file.group.ljust(max_group),
+        file.bytesize.to_s.rjust(max_size),
+        file.mtime.strftime('%-m月').rjust(5),
+        file.mtime.strftime('%e %H:%M'),
```

Comment:
1行にまとめることはできないですかね〜？

```suggestion
        file.mtime.strftime(????),
```

Reply:
こちら１桁の月と２桁の月をきれいに揃えるのが難しいと判断し、操作を分けておりましたが、１行で記載することも可能です。
方法として、この２行を連結する、もしくは`format`(`sprintf`)で揃えてあげる、という２種類があるかと思います。
今回は`sprintf`を利用して１行で出力するようにしてみましたが、この修正でよろしいでしょうか？